src/services/organizationService.ts

```diff
@@ -0,0 +1,86 @@
+import { Op } from "sequelize";
+import { IpaPublicAdministration } from "../models/IpaPublicAdministration";
+import { Organization } from "../models/Organization";
+import { User } from "../models/User";
+import {
+  fromOrganizationModelToSearchedOrganization,
+  fromPublicAdministrationToSearchedOrganization,
+  ISearchedOrganization
+} from "../types/organization";
+
+/**
+ * Retrieve from the db all the public administrations whose names match the provided value.
+ * In order to match, the name of the public administration must include each word of the input value in the same order.
+ * @param input The value to compare with the name of the public administration.
+ */
+export async function findPublicAdministrationsByName(
+  input: string
+): Promise<ReadonlyArray<ISearchedOrganization>> {
+  const descriptionWords = input
+    .split(" ")
+    .reduce(
+      (words: ReadonlyArray<string>, word: string) =>
+        word ? words.concat(word) : words,
+      []
+    );
+  const organizations = await Organization.findAll({
+    include: [
+      {
+        as: "legalRepresentative",
+        model: User
+      }
+    ],
+    where: {
+      name: {
+        [Op.iLike]: `%${descriptionWords.join("%")}%`
```

Comment:
have you heard anything about sql injection? you're taking a raw query parameter and feeding it to an sql query as is!

btw we're talking about a few thousand records here, why don't we do the search on the client? it's going to be much more efficient

on a side note, postgres provides [full text search indexes](https://dzone.com/articles/postgres-full-text-search), much better that doing a LIKE query

Reply:
Currently the PAs on IPA are 22729, which is not a huge amount, but the payload of the response could be pretty big if the whole information for each PA is sent.
So I think it could be better if only the needed attributes for the PA from IPA are returned (i.e. the name and the IPA code which are needed to populate a dropdown) and then another endpoint will be created, which will return all the attributes for the specific PA selected from the dropdown.
Is this a good approach? About the required attributes for the first call, should they be sent in the request as query string parameters?

Reply:
in that case (volume) yeah the choice of keeping them in the server is fine - my concern is that the query as you wrote it is very inefficient (it's basically a linear search), so take a look at full text indexes in postgres

the step that is missing is that you need to validate the content of the strings you feed to the database, you need [to sanitize it](https://www.quora.com/What-exactly-is-data-sanitization-with-respect-to-SQL-injection), I'm sure there's some js library that does that for you